handler/handler_test.go

```diff
@@ -619,3 +599,30 @@ func TestWipeMetricStore(t *testing.T) {
 		}
 	}
 }
+
+// verifyMetricFamily jumps through a few hoops because the current protobuf
+// implementation is deliberately creating an unstable formatting for the text
+// representation. So this takes the text representation of the expected
+// MetricFamily and unmarshals it into a proto message object first. Then it
+// marshals both the expected and the got proto message into a binary protobuf,
+// which it then compares.
+func verifyMetricFamily(t *testing.T, expText string, got *dto.MetricFamily) {
+	gotProto, err := proto.Marshal(got)
+	if err != nil {
+		t.Errorf("unexpected error marshaling MetricFamily %v", got)
+	}
+
+	exp := &dto.MetricFamily{}
+	err = proto.UnmarshalText(expText, exp)
```

Comment:
This can be updated to use `google.golang.org/protobuf/encoding/prototext`.

```suggestion
	err = prototext.Unmarshal([]byte(expText), exp)
```